h2o-r/h2o-package/R/automl.R

```diff
@@ -580,3 +581,85 @@ h2o.get_leaderboard <- function(object, extra_columns=NULL) {
   if (!.is.H2OAutoML(object)) stop("Only H2OAutoML instances are currently supported.")
   return(.automl.fetch_leaderboard(attr(object, 'id'), extra_columns))
 }
+
+
+#' Get best model of a given family/algorithm for a given criterion from an AutoML object.
+#'
+#' @param object H2OAutoML object
+#' @param algorithm One of "any", "basemodel", "deeplearning", "drf", "gbm", "glm", "stackedensemble", "xgboost"
+#' @param criterion Criterion can be one of the metrics reported in the leaderboard. If set to NULL, the same ordering
+#' as in the leaderboard will be used.
+#' Avaliable criteria:
+#' \itemize{
+#' \item{Regression metrics: deviance, RMSE, MSE, MAE, RMSLE}
+#' \item{Binomial metrics: AUC, logloss, AUCPR, mean_per_class_error, RMSE, MSE}
+#' \item{Multinomial metrics: mean_per_class_error, logloss, RMSE, MSE, AUC, AUCPR}
+#' }
+#' The following additional leaderboard information can be also used as a criterion:
+#' \itemize{
+#' \item{'training_time_ms': column providing the training time of each model in milliseconds (doesn't include the training of cross validation models).}
+#' \item{'predict_time_per_row_ms': column providing the average prediction time by the model for a single row.}
+#' }
+#' @return An H2OModel or NULL if no model of a given family is present
+#' @examples
+#' \dontrun{
+#' library(h2o)
+#' h2o.init()
+#' prostate_path <- system.file("extdata", "prostate.csv", package = "h2o")
+#' prostate <- h2o.importFile(path = prostate_path, header = TRUE)
+#' y <- "CAPSULE"
+#' prostate[,y] <- as.factor(prostate[,y])  #convert to factor for classification
+#' aml <- h2o.automl(y = y, training_frame = prostate, max_runtime_secs = 30)
+#' gbm <- h2o.get_best_model(aml, "gbm")
+#' }
+#' @export
+h2o.get_best_model <- function(object,
+                               algorithm = c("any", "basemodel", "deeplearning", "drf", "gbm",
+                                             "glm", "stackedensemble", "xgboost"),
+                               criterion = c("AUTO", "AUC", "AUCPR", "logloss", "MAE", "mean_per_class_error",
+                                             "deviance", "MSE", "predict_time_per_row_ms",
+                                             "RMSE", "RMSLE", "training_time_ms")) {
```

Comment:
is it  also possible to externalize some of those collections in R so that we can reuse them?
```R
.h2o.automl_supported_algos = c("deeplearning", "drf", "gbm", "glm", "stackedensemble", "xgboost")
...
     algorithm = c("any", "basemodel", .h2o.automl_supported_algos)
```
and something similar with the metrics/leaderboard columns.
This would allow reuse in `h2o.automl` logic as well.